lib/Reader/readerir.cpp

```diff
@@ -6543,6 +6543,40 @@ IRNode *GenIR::loadAndBox(CORINFO_RESOLVED_TOKEN *ResolvedToken, IRNode *Addr,
   return box(ResolvedToken, Value);
 }
 
+IRNode *GenIR::makeRefAny(CORINFO_RESOLVED_TOKEN *ResolvedToken,
+                          IRNode *Object) {
+
+  // Create a new temporary of the right type.
+  CORINFO_CLASS_HANDLE RefAnyHandle = getBuiltinClass(CLASSID_TYPED_BYREF);
+  Type *RefAnyTy = getType(CORINFO_TYPE_VALUECLASS, RefAnyHandle);
+  assert(RefAnyTy->isStructTy() && "expect struct type for refany");
+  StructType *RefAnyStructTy = cast<StructType>(RefAnyTy);
+  Value *RefAny = createTemporary(RefAnyTy, "RefAny");
+  const bool IsVolatile = false;
+
+  // Store the object in the value field. Object should either be an object
+  // reference or the address of a local or param, both of which we type as
+  // managed pointers.
+  assert(isManagedPointerType(Object->getType()) &&
+         "wrong type for refany value");
+  Value *ObjectFieldAddress = LLVMBuilder->CreateStructGEP(RefAnyTy, RefAny, 0);
```

Comment:
I would make 0 and 1 symbolic constants here (e.g., ObjectFieldIndex and TypeFieldIndex)